Kernel/Devices/SB16.cpp

```diff
@@ -74,27 +72,19 @@ UNMAP_AFTER_INIT SB16::~SB16()
 {
 }
 
-UNMAP_AFTER_INIT void SB16::detect()
+UNMAP_AFTER_INIT RefPtr<SB16> SB16::try_detect()
```

Comment:
`try_detect` is a little misleading, this is not trying to detect the SB16, this is trying to create an SB16 instance (but could fail if the hardware device was not found), can we come up with a better name?

Reply:
It does try to detect SB16, but yeah, I think a better name would be `try_detect_and_create` here.

Reply:
That function does the detect per what is a normal SB16 supposed to signal to the driver upon the end of detection procedure: https://wiki.osdev.org/Sound_Blaster_16#Reseting_DSP_.28detecting_DSP.29